src/System/extract_system_rep.rs

```diff
@@ -40,21 +41,34 @@ pub fn create_executable_query<'a>(
             QueryExpression::Reachability(automata, start, end) => {
                 let mut quotient_index = None;
                 let machine = get_system_recipe(automata, component_loader, &mut dim, &mut quotient_index);
-                if let Err(e) = validate_reachability_input(&machine, start, end){
+
+                if let Err(e) = validate_reachability_input(&machine, end){
                     return Err(e.into());
                 }
+
                 let transition_system = machine.clone().compile(dim)?;
 
+                let start_state: State = if let Some(state) = &**start {
```

Comment:
maybe use a match instead of if let some, as we check all arms anyway

Reply:
Fixed og ser hot ud nu